mssqlcli/main.py

```diff
@@ -623,13 +559,38 @@ def _handle_server_closed_connection(self):
             show_default=False, type=bool, default=True)
         if reconnect:
             try:
-                reset_connection_and_clients(self.sqltoolsclient,
-                                             self.mssqlcliclient_query_execution)
+                self.reset_connection()
 
                 click.secho('Reconnected!\nTry the command again.', fg='green')
             except Exception as e:
                 click.secho(str(e), err=True, fg='red')
 
+    def reset_connection(self):
+        """
+        Reset mssqlcli client with a new sql tools service and connection.
+        """
+        try:
+            self.sqltoolsclient.shutdown()
+            self.sqltoolsclient = SqlToolsClient()
+
+            self.mssqlcliclient_main.sql_tools_client = self.sqltoolsclient
+            self.mssqlcliclient_main.is_connected = False
+            self.mssqlcliclient_main.owner_uri = generate_owner_uri()
+

```

Comment:
I feel like we should clone the existing mssqlclient such that we know we're in a clean state.

Reply:
agreed, I like that idea. Will add it next.

---
In reply to: [170394199](https://github.com/dbcli/mssql-cli/pull/170#discussion_r170394199) [](ancestors = 170394199)